### PR TITLE
Create Offers Mutation

### PR DESCRIPTION
**Pull request by dcoleman21**
## Description

Create offers table
Create offers mutation
Create offers type
Write happy and sad path testing for offer mutations
## How Has This Been Tested?

Model testing and Happy and Sad Path testing for create_offer request

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] I have updated the README to reflect any changes.

### CODE REVIEW

File: app/graphql/mutations/create_offer.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
+module Mutations
+  class CreateOffer < Mutations::BaseMutation
+    description "Create a offer record"
+
+    argument :user_id, Integer, required: false
```
Comment (Callbritton) [inline app/graphql/mutations/create_offer.rb:5]: This looks great! Thank you :)

**Review by Callbritton — APPROVED** [review]
Excellent job!

File: app/graphql/types/offer_type.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+module Types
+  class OfferType < Types::BaseObject
+    field :user, Types::UserType, null: false
+    field :listing, Types::ListingType, null: false
```
Comment (jdcarey128) [inline app/graphql/types/offer_type.rb:4]: Nice job on these nested types!

**Review by jdcarey128 — APPROVED** [review]
This looks great team. Nice work!

**Pull request merged**